contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6180,6 +6183,36 @@ transformSelectIntoStmt(CreateTableAsStmt *stmt)
 		foreach (elements, q->targetList)
 		{
 			TargetEntry *tle = (TargetEntry *)lfirst(elements);
+			char *original_name = NULL;
+
+			if (tle->resname != NULL && !tle->resjunk)
+			{	
+
+				original_name = tle->resname;
+				tle->resname = downcase_identifier(tle->resname, strlen(tle->resname), false, false);
+
+				/* check if the original_name is already in lowercase */
+				if (original_name != NULL && strcmp(original_name, tle->resname) && strlen(tle->resname) < NAMEDATALEN - 1)
```

Comment:
I suspect `strlen(tle->resname) < NAMEDATALEN - 1` might differ for multi-bytes characters, let's check with @Deepesh125 regarding it.